src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurExpenseV3ServiceImpl.java

```diff
@@ -144,11 +153,47 @@ protected void validateExpenseAllocations(String accessToken, List<ConcurEventNo
         
     }
     
-    
     protected void updateStatusInConcur(String accessToken, String reportId, boolean reportValid, ConcurEventNotificationProcessingResultsDTO resultsDTO) {
         LOG.info("updateStatusInConcur, for report id " + reportId + " the over all validation status will be set to " + reportValid);
+        if (!shouldUpdateStatusInConcur()) {
+            LOG.info("updateStatusInConcur, Concur workflow actions are currently disabled in this KFS environment");
+            return;
+        }
+
+        String workflowAction = reportValid ? ConcurWorkflowActions.APPROVE : ConcurWorkflowActions.SEND_BACK;
+        String workflowActionUrl = buildFullUrlForExpenseWorkflowAction(reportId, workflowAction);
+        String workflowComment = reportValid
+                ? ConcurConstants.APPROVE_COMMENT
+                : ConcurUtils.buildValidationErrorMessageForWorkflowAction(resultsDTO);
+        ConcurV4WorkflowDTO workflowDTO = new ConcurV4WorkflowDTO(workflowComment);
+        String expenseV4WorkflowMessageFormat = configurationService.getPropertyValueAsString(
+                ConcurKeyConstants.MESSAGE_CONCUR_EXPENSEV4_EXPENSE_REPORT_WORKFLOW);
+        String logMessageDetail = MessageFormat.format(expenseV4WorkflowMessageFormat, workflowAction, reportId);
+        
+        ConcurWebRequest<Void> webRequest = ConcurWebRequestBuilder.forRequestExpectingEmptyResponse()
+                .withUrl(workflowActionUrl)
+                .withHttpMethod(HttpMethod.PATCH)
+                .withJsonBody(workflowDTO)
+                .build();
         
-        LOG.info("updateStatusInConcur, update of status in Concur not implemented yet");
+        concurEventNotificationV2WebserviceService.callConcurEndpoint(
```

Comment:
updateStatusInConcur is somewhat of a long method. Would abstracting the building of the web request make the method shorter and cleaner?

Reply:
I'll look into refactoring this method to make it shorter.